test/jdk/java/nio/file/Files/Misc.java

```diff
@@ -87,6 +87,13 @@ static void testCreateDirectories(Path tmpdir) throws IOException {
      * Tests isHidden
      */
     static void testIsHidden(Path tmpdir) throws IOException {
+        // an empty path must not throw an AIOOB
+        try {
+            isHidden(Path.of(""));
+        } catch (ArrayIndexOutOfBoundsException exception) {
+            throw new AssertionError("8255576", exception);
+        }
+
```

Comment:
Thanks for taking this one. The implementation changes are good.

For the test then I don't think we need to catch AIOOB, the test should fail if any exception is thrown. I think think of any scenario where it would be sensible to run the tests when in a hidden directory so we could change this to test that isHidden returns false - do you agree?

Reply:
> For the test then I don't think we need to catch AIOOB, the test should fail if any exception is thrown.

Okay, I'll change the catch clause to throw on any `RuntimeException` caught. Or is it also fine to omit the try-catch statements and not provide a specific failure message?

> ... scenario where it would be sensible to run the tests when in a hidden directory ...

What about testing that new scenario additionally to the plain one passing an empty `Path`?

That new scenario would include:
  - create a hidden directory in platform-agnostic manner
  - start a test program using this directory as the working directory
  - assert (by checking the output of the test program?) isHidden returned false

Reply:
Thanks for the update, I think I would put an assertFalse around this because the current directory shouldn't be hidden.

Creating a new VM that runs in a hidden directory is probably a stretch. I don't think it is strictly required here but don't object to reviewing any proposals if you are interested in spending time on it.

Reply:
I wrapped a "logical assertFalse" around the call -- there were already 32 cases of `assertTrue(! ...)` via https://github.com/openjdk/jdk/pull/952/commits/c27abeab3a38daf2ae5dc25894c8b966d9edfc58

I want to defer such a check to a different issue. If there was a feature like "chroot(Path)", or a virtual file system that could be mapped to the default one so that Path("") resolves to a hidden working directory ... or something else that's better (and faster) than spawning a new VM.

Reply:
This test pre-dates the use of TestNG support so had its own assertTrue. We could add an assertFalse but what you have is okay too as it's just one usage.

A chroot(Path) like API would break the assumption that the current directory doesn't change. Every relative path would locate a different file. So if we add a test, and I don't think it's too important, then it would need to launch a new VM. There is test lib infrastructure to support this. Historically, multi VM tests tended to have reliability issues, also they are significantly slower that tests that run "in-process".